diff --git a/lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp b/lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp
index 10e9e498fa63..17462acd273f 100644
--- a/lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp
+++ b/lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp
@@ -131,8 +131,8 @@ swizzleIndexes(ConversionPatternRewriter &rewriter, Location loc, Value row,
  * @param reps number of instructions repretition to fully cover dot operand
  * @param smemStrides strides in LDS tensor
  * @param loadVecSize number of elements loaded by one operation
- * @param iNonKDim non-K dimension of dot operand
- * @param iKDim non-K dimension of dot operand
+ * @param iNonKDim non-K dimension size of one MFMA instruction
+ * @param iKDim K dimension size of one MFMA instruction
  * @return vector (i-th element corresponds to i-th load instruction) of
  * 2-element vectors(tensor row and col).
  */
diff --git a/lib/Dialect/TritonGPU/IR/Dialect.cpp b/lib/Dialect/TritonGPU/IR/Dialect.cpp
index d3e5673f97a2..d9011f58a4b0 100644
--- a/lib/Dialect/TritonGPU/IR/Dialect.cpp
+++ b/lib/Dialect/TritonGPU/IR/Dialect.cpp
@@ -1434,8 +1434,8 @@ Attribute MfmaEncodingAttr::parse(AsmParser &parser, Type type) {
 
 void MfmaEncodingAttr::print(AsmPrinter &printer) const {
   printer << "<{"
-          << "version = " << getVersionMajor() << "." << getVersionMinor()
-          << ", "
+          << "versionMajor = " << getVersionMajor() << ", "
+          << "versionMinor = " << getVersionMinor() << ", "
           << "warpsPerCTA = [" << getWarpsPerCTA() << "], "
           << "instrShape = [" << getMDim() << ", " << getNDim() << "], "
           << "isTransposed = " << getIsTransposed() << "}>";
diff --git a/test/TritonGPU/accelerate-matmul-cdna1.mlir b/test/TritonGPU/accelerate-matmul-cdna1.mlir
index 9b1f635d17ca..51956c590035 100644
--- a/test/TritonGPU/accelerate-matmul-cdna1.mlir
+++ b/test/TritonGPU/accelerate-matmul-cdna1.mlir
@@ -7,7 +7,7 @@
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
 // CHECK: convert_dot_32_32_32_f16_f16_f32
     tt.func @convert_dot_32_32_32_f16_f16_f32(%a: tensor<32x32x!a_ty, #dot_operand_a>, %b: tensor<32x32x!b_ty, #dot_operand_b>) -> tensor<32x32x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<32x32x!c_ty, #blocked>
@@ -28,7 +28,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
 // CHECK: convert_dot_32_32_32_bf16_bf16_f32
     tt.func @convert_dot_32_32_32_bf16_bf16_f32(%a: tensor<32x32x!a_ty, #dot_operand_a>, %b: tensor<32x32x!b_ty, #dot_operand_b>) -> tensor<32x32x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<32x32x!c_ty, #blocked>
@@ -49,7 +49,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
 // CHECK: convert_dot_32_32_32_f32_f32_f32
     tt.func @convert_dot_32_32_32_f32_f32_f32(%a: tensor<32x32x!a_ty, #dot_operand_a>, %b: tensor<32x32x!b_ty, #dot_operand_b>) -> tensor<32x32x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<32x32x!c_ty, #blocked>
@@ -70,7 +70,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
 // CHECK: convert_dot_32_32_32_i8_i8_i32
     tt.func @convert_dot_32_32_32_i8_i8_i32(%a: tensor<32x32x!a_ty, #dot_operand_a>, %b: tensor<32x32x!b_ty, #dot_operand_b>) -> tensor<32x32x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0> : tensor<32x32x!c_ty, #blocked>
@@ -167,7 +167,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
 // CHECK: convert_dot_16_16_32_f16_f16_f32
     tt.func @convert_dot_16_16_32_f16_f16_f32(%a: tensor<16x32x!a_ty, #dot_operand_a>, %b: tensor<32x16x!b_ty, #dot_operand_b>) -> tensor<16x16x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<16x16x!c_ty, #blocked>
@@ -188,7 +188,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
 // CHECK: convert_dot_16_16_32_bf16_bf16_f32
     tt.func @convert_dot_16_16_32_bf16_bf16_f32(%a: tensor<16x32x!a_ty, #dot_operand_a>, %b: tensor<32x16x!b_ty, #dot_operand_b>) -> tensor<16x16x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<16x16x!c_ty, #blocked>
@@ -209,7 +209,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
 // CHECK: convert_dot_16_16_32_f32_f32_f32
     tt.func @convert_dot_16_16_32_f32_f32_f32(%a: tensor<16x32x!a_ty, #dot_operand_a>, %b: tensor<32x16x!b_ty, #dot_operand_b>) -> tensor<16x16x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<16x16x!c_ty, #blocked>
@@ -230,7 +230,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
 // CHECK: convert_dot_16_16_32_i8_i8_i32
     tt.func @convert_dot_16_16_32_i8_i8_i32(%a: tensor<16x32x!a_ty, #dot_operand_a>, %b: tensor<32x16x!b_ty, #dot_operand_b>) -> tensor<16x16x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0> : tensor<16x16x!c_ty, #blocked>
@@ -327,7 +327,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
 // CHECK: convert_dot_4_4_64_f16_f16_f32
     tt.func @convert_dot_4_4_64_f16_f16_f32(%a: tensor<4x64x!a_ty, #dot_operand_a>, %b: tensor<64x4x!b_ty, #dot_operand_b>) -> tensor<4x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x4x!c_ty, #blocked>
@@ -348,7 +348,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
 // CHECK: convert_dot_4_4_64_bf16_bf16_f32
     tt.func @convert_dot_4_4_64_bf16_bf16_f32(%a: tensor<4x64x!a_ty, #dot_operand_a>, %b: tensor<64x4x!b_ty, #dot_operand_b>) -> tensor<4x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x4x!c_ty, #blocked>
@@ -369,7 +369,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
 // CHECK: convert_dot_4_4_64_f32_f32_f32
     tt.func @convert_dot_4_4_64_f32_f32_f32(%a: tensor<4x64x!a_ty, #dot_operand_a>, %b: tensor<64x4x!b_ty, #dot_operand_b>) -> tensor<4x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x4x!c_ty, #blocked>
@@ -390,7 +390,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
 // CHECK: convert_dot_4_4_64_i8_i8_i32
     tt.func @convert_dot_4_4_64_i8_i8_i32(%a: tensor<4x64x!a_ty, #dot_operand_a>, %b: tensor<64x4x!b_ty, #dot_operand_b>) -> tensor<4x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0> : tensor<4x4x!c_ty, #blocked>
@@ -487,7 +487,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
 // CHECK: convert_dot_64_4_4_f16_f16_f32
     tt.func @convert_dot_64_4_4_f16_f16_f32(%a: tensor<64x4x!a_ty, #dot_operand_a>, %b: tensor<4x4x!b_ty, #dot_operand_b>) -> tensor<64x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<64x4x!c_ty, #blocked>
@@ -508,7 +508,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
 // CHECK: convert_dot_64_4_4_bf16_bf16_f32
     tt.func @convert_dot_64_4_4_bf16_bf16_f32(%a: tensor<64x4x!a_ty, #dot_operand_a>, %b: tensor<4x4x!b_ty, #dot_operand_b>) -> tensor<64x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<64x4x!c_ty, #blocked>
@@ -529,7 +529,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
 // CHECK: convert_dot_64_4_4_f32_f32_f32
     tt.func @convert_dot_64_4_4_f32_f32_f32(%a: tensor<64x4x!a_ty, #dot_operand_a>, %b: tensor<4x4x!b_ty, #dot_operand_b>) -> tensor<64x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<64x4x!c_ty, #blocked>
@@ -550,7 +550,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
 // CHECK: convert_dot_64_4_4_i8_i8_i32
     tt.func @convert_dot_64_4_4_i8_i8_i32(%a: tensor<64x4x!a_ty, #dot_operand_a>, %b: tensor<4x4x!b_ty, #dot_operand_b>) -> tensor<64x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0> : tensor<64x4x!c_ty, #blocked>
@@ -647,7 +647,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
 // CHECK: convert_dot_4_64_4_f16_f16_f32
     tt.func @convert_dot_4_64_4_f16_f16_f32(%a: tensor<4x4x!a_ty, #dot_operand_a>, %b: tensor<4x64x!b_ty, #dot_operand_b>) -> tensor<4x64x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x64x!c_ty, #blocked>
@@ -668,7 +668,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
 // CHECK: convert_dot_4_64_4_bf16_bf16_f32
     tt.func @convert_dot_4_64_4_bf16_bf16_f32(%a: tensor<4x4x!a_ty, #dot_operand_a>, %b: tensor<4x64x!b_ty, #dot_operand_b>) -> tensor<4x64x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x64x!c_ty, #blocked>
@@ -689,7 +689,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
 // CHECK: convert_dot_4_64_4_f32_f32_f32
     tt.func @convert_dot_4_64_4_f32_f32_f32(%a: tensor<4x4x!a_ty, #dot_operand_a>, %b: tensor<4x64x!b_ty, #dot_operand_b>) -> tensor<4x64x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x64x!c_ty, #blocked>
@@ -710,7 +710,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 1.0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
 // CHECK: convert_dot_4_64_4_i8_i8_i32
     tt.func @convert_dot_4_64_4_i8_i8_i32(%a: tensor<4x4x!a_ty, #dot_operand_a>, %b: tensor<4x64x!b_ty, #dot_operand_b>) -> tensor<4x64x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0> : tensor<4x64x!c_ty, #blocked>
diff --git a/test/TritonGPU/accelerate-matmul-cdna2.mlir b/test/TritonGPU/accelerate-matmul-cdna2.mlir
index c1e4700f554f..0d853186bac3 100644
--- a/test/TritonGPU/accelerate-matmul-cdna2.mlir
+++ b/test/TritonGPU/accelerate-matmul-cdna2.mlir
@@ -7,7 +7,7 @@
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
 // CHECK: convert_dot_32_32_32_f16_f16_f32
     tt.func @convert_dot_32_32_32_f16_f16_f32(%a: tensor<32x32x!a_ty, #dot_operand_a>, %b: tensor<32x32x!b_ty, #dot_operand_b>) -> tensor<32x32x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<32x32x!c_ty, #blocked>
@@ -28,7 +28,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
 // CHECK: convert_dot_32_32_32_bf16_bf16_f32
     tt.func @convert_dot_32_32_32_bf16_bf16_f32(%a: tensor<32x32x!a_ty, #dot_operand_a>, %b: tensor<32x32x!b_ty, #dot_operand_b>) -> tensor<32x32x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<32x32x!c_ty, #blocked>
@@ -49,7 +49,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
 // CHECK: convert_dot_32_32_32_f32_f32_f32
     tt.func @convert_dot_32_32_32_f32_f32_f32(%a: tensor<32x32x!a_ty, #dot_operand_a>, %b: tensor<32x32x!b_ty, #dot_operand_b>) -> tensor<32x32x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<32x32x!c_ty, #blocked>
@@ -70,7 +70,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
 // CHECK: convert_dot_32_32_32_i8_i8_i32
     tt.func @convert_dot_32_32_32_i8_i8_i32(%a: tensor<32x32x!a_ty, #dot_operand_a>, %b: tensor<32x32x!b_ty, #dot_operand_b>) -> tensor<32x32x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0> : tensor<32x32x!c_ty, #blocked>
@@ -167,7 +167,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
 // CHECK: convert_dot_16_16_32_f16_f16_f32
     tt.func @convert_dot_16_16_32_f16_f16_f32(%a: tensor<16x32x!a_ty, #dot_operand_a>, %b: tensor<32x16x!b_ty, #dot_operand_b>) -> tensor<16x16x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<16x16x!c_ty, #blocked>
@@ -188,7 +188,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
 // CHECK: convert_dot_16_16_32_bf16_bf16_f32
     tt.func @convert_dot_16_16_32_bf16_bf16_f32(%a: tensor<16x32x!a_ty, #dot_operand_a>, %b: tensor<32x16x!b_ty, #dot_operand_b>) -> tensor<16x16x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<16x16x!c_ty, #blocked>
@@ -209,7 +209,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
 // CHECK: convert_dot_16_16_32_f32_f32_f32
     tt.func @convert_dot_16_16_32_f32_f32_f32(%a: tensor<16x32x!a_ty, #dot_operand_a>, %b: tensor<32x16x!b_ty, #dot_operand_b>) -> tensor<16x16x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<16x16x!c_ty, #blocked>
@@ -230,7 +230,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
 // CHECK: convert_dot_16_16_32_i8_i8_i32
     tt.func @convert_dot_16_16_32_i8_i8_i32(%a: tensor<16x32x!a_ty, #dot_operand_a>, %b: tensor<32x16x!b_ty, #dot_operand_b>) -> tensor<16x16x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0> : tensor<16x16x!c_ty, #blocked>
@@ -327,7 +327,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
 // CHECK: convert_dot_4_4_64_f16_f16_f32
     tt.func @convert_dot_4_4_64_f16_f16_f32(%a: tensor<4x64x!a_ty, #dot_operand_a>, %b: tensor<64x4x!b_ty, #dot_operand_b>) -> tensor<4x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x4x!c_ty, #blocked>
@@ -348,7 +348,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
 // CHECK: convert_dot_4_4_64_bf16_bf16_f32
     tt.func @convert_dot_4_4_64_bf16_bf16_f32(%a: tensor<4x64x!a_ty, #dot_operand_a>, %b: tensor<64x4x!b_ty, #dot_operand_b>) -> tensor<4x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x4x!c_ty, #blocked>
@@ -369,7 +369,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
 // CHECK: convert_dot_4_4_64_f32_f32_f32
     tt.func @convert_dot_4_4_64_f32_f32_f32(%a: tensor<4x64x!a_ty, #dot_operand_a>, %b: tensor<64x4x!b_ty, #dot_operand_b>) -> tensor<4x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x4x!c_ty, #blocked>
@@ -390,7 +390,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
 // CHECK: convert_dot_4_4_64_i8_i8_i32
     tt.func @convert_dot_4_4_64_i8_i8_i32(%a: tensor<4x64x!a_ty, #dot_operand_a>, %b: tensor<64x4x!b_ty, #dot_operand_b>) -> tensor<4x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0> : tensor<4x4x!c_ty, #blocked>
@@ -487,7 +487,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
 // CHECK: convert_dot_64_4_4_f16_f16_f32
     tt.func @convert_dot_64_4_4_f16_f16_f32(%a: tensor<64x4x!a_ty, #dot_operand_a>, %b: tensor<4x4x!b_ty, #dot_operand_b>) -> tensor<64x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<64x4x!c_ty, #blocked>
@@ -508,7 +508,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
 // CHECK: convert_dot_64_4_4_bf16_bf16_f32
     tt.func @convert_dot_64_4_4_bf16_bf16_f32(%a: tensor<64x4x!a_ty, #dot_operand_a>, %b: tensor<4x4x!b_ty, #dot_operand_b>) -> tensor<64x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<64x4x!c_ty, #blocked>
@@ -529,7 +529,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
 // CHECK: convert_dot_64_4_4_f32_f32_f32
     tt.func @convert_dot_64_4_4_f32_f32_f32(%a: tensor<64x4x!a_ty, #dot_operand_a>, %b: tensor<4x4x!b_ty, #dot_operand_b>) -> tensor<64x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<64x4x!c_ty, #blocked>
@@ -550,7 +550,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
 // CHECK: convert_dot_64_4_4_i8_i8_i32
     tt.func @convert_dot_64_4_4_i8_i8_i32(%a: tensor<64x4x!a_ty, #dot_operand_a>, %b: tensor<4x4x!b_ty, #dot_operand_b>) -> tensor<64x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0> : tensor<64x4x!c_ty, #blocked>
@@ -647,7 +647,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
 // CHECK: convert_dot_4_64_4_f16_f16_f32
     tt.func @convert_dot_4_64_4_f16_f16_f32(%a: tensor<4x4x!a_ty, #dot_operand_a>, %b: tensor<4x64x!b_ty, #dot_operand_b>) -> tensor<4x64x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x64x!c_ty, #blocked>
@@ -668,7 +668,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
 // CHECK: convert_dot_4_64_4_bf16_bf16_f32
     tt.func @convert_dot_4_64_4_bf16_bf16_f32(%a: tensor<4x4x!a_ty, #dot_operand_a>, %b: tensor<4x64x!b_ty, #dot_operand_b>) -> tensor<4x64x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x64x!c_ty, #blocked>
@@ -689,7 +689,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
 // CHECK: convert_dot_4_64_4_f32_f32_f32
     tt.func @convert_dot_4_64_4_f32_f32_f32(%a: tensor<4x4x!a_ty, #dot_operand_a>, %b: tensor<4x64x!b_ty, #dot_operand_b>) -> tensor<4x64x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x64x!c_ty, #blocked>
@@ -710,7 +710,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 2.0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 2, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
 // CHECK: convert_dot_4_64_4_i8_i8_i32
     tt.func @convert_dot_4_64_4_i8_i8_i32(%a: tensor<4x4x!a_ty, #dot_operand_a>, %b: tensor<4x64x!b_ty, #dot_operand_b>) -> tensor<4x64x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0> : tensor<4x64x!c_ty, #blocked>
diff --git a/test/TritonGPU/accelerate-matmul-cdna3.mlir b/test/TritonGPU/accelerate-matmul-cdna3.mlir
index 2d0f94f73f9a..24d8ee993615 100644
--- a/test/TritonGPU/accelerate-matmul-cdna3.mlir
+++ b/test/TritonGPU/accelerate-matmul-cdna3.mlir
@@ -7,7 +7,7 @@
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
 // CHECK: convert_dot_32_32_32_f16_f16_f32
     tt.func @convert_dot_32_32_32_f16_f16_f32(%a: tensor<32x32x!a_ty, #dot_operand_a>, %b: tensor<32x32x!b_ty, #dot_operand_b>) -> tensor<32x32x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<32x32x!c_ty, #blocked>
@@ -28,7 +28,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
 // CHECK: convert_dot_32_32_32_bf16_bf16_f32
     tt.func @convert_dot_32_32_32_bf16_bf16_f32(%a: tensor<32x32x!a_ty, #dot_operand_a>, %b: tensor<32x32x!b_ty, #dot_operand_b>) -> tensor<32x32x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<32x32x!c_ty, #blocked>
@@ -49,7 +49,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
 // CHECK: convert_dot_32_32_32_f32_f32_f32
     tt.func @convert_dot_32_32_32_f32_f32_f32(%a: tensor<32x32x!a_ty, #dot_operand_a>, %b: tensor<32x32x!b_ty, #dot_operand_b>) -> tensor<32x32x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<32x32x!c_ty, #blocked>
@@ -70,7 +70,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
 // CHECK: convert_dot_32_32_32_i8_i8_i32
     tt.func @convert_dot_32_32_32_i8_i8_i32(%a: tensor<32x32x!a_ty, #dot_operand_a>, %b: tensor<32x32x!b_ty, #dot_operand_b>) -> tensor<32x32x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0> : tensor<32x32x!c_ty, #blocked>
@@ -91,7 +91,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
 // CHECK: convert_dot_32_32_32_f8E4M3FNUZ_f8E4M3FNUZ_f32
     tt.func @convert_dot_32_32_32_f8E4M3FNUZ_f8E4M3FNUZ_f32(%a: tensor<32x32x!a_ty, #dot_operand_a>, %b: tensor<32x32x!b_ty, #dot_operand_b>) -> tensor<32x32x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<32x32x!c_ty, #blocked>
@@ -112,7 +112,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
 // CHECK: convert_dot_32_32_32_f8E4M3FNUZ_f8E5M2FNUZ_f32
     tt.func @convert_dot_32_32_32_f8E4M3FNUZ_f8E5M2FNUZ_f32(%a: tensor<32x32x!a_ty, #dot_operand_a>, %b: tensor<32x32x!b_ty, #dot_operand_b>) -> tensor<32x32x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<32x32x!c_ty, #blocked>
@@ -133,7 +133,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
 // CHECK: convert_dot_32_32_32_f8E5M2FNUZ_f8E4M3FNUZ_f32
     tt.func @convert_dot_32_32_32_f8E5M2FNUZ_f8E4M3FNUZ_f32(%a: tensor<32x32x!a_ty, #dot_operand_a>, %b: tensor<32x32x!b_ty, #dot_operand_b>) -> tensor<32x32x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<32x32x!c_ty, #blocked>
@@ -154,7 +154,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [32, 32], isTransposed = false}>
 // CHECK: convert_dot_32_32_32_f8E5M2FNUZ_f8E5M2FNUZ_f32
     tt.func @convert_dot_32_32_32_f8E5M2FNUZ_f8E5M2FNUZ_f32(%a: tensor<32x32x!a_ty, #dot_operand_a>, %b: tensor<32x32x!b_ty, #dot_operand_b>) -> tensor<32x32x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<32x32x!c_ty, #blocked>
@@ -175,7 +175,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
 // CHECK: convert_dot_16_16_32_f16_f16_f32
     tt.func @convert_dot_16_16_32_f16_f16_f32(%a: tensor<16x32x!a_ty, #dot_operand_a>, %b: tensor<32x16x!b_ty, #dot_operand_b>) -> tensor<16x16x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<16x16x!c_ty, #blocked>
@@ -196,7 +196,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
 // CHECK: convert_dot_16_16_32_bf16_bf16_f32
     tt.func @convert_dot_16_16_32_bf16_bf16_f32(%a: tensor<16x32x!a_ty, #dot_operand_a>, %b: tensor<32x16x!b_ty, #dot_operand_b>) -> tensor<16x16x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<16x16x!c_ty, #blocked>
@@ -217,7 +217,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
 // CHECK: convert_dot_16_16_32_f32_f32_f32
     tt.func @convert_dot_16_16_32_f32_f32_f32(%a: tensor<16x32x!a_ty, #dot_operand_a>, %b: tensor<32x16x!b_ty, #dot_operand_b>) -> tensor<16x16x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<16x16x!c_ty, #blocked>
@@ -238,7 +238,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
 // CHECK: convert_dot_16_16_32_i8_i8_i32
     tt.func @convert_dot_16_16_32_i8_i8_i32(%a: tensor<16x32x!a_ty, #dot_operand_a>, %b: tensor<32x16x!b_ty, #dot_operand_b>) -> tensor<16x16x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0> : tensor<16x16x!c_ty, #blocked>
@@ -259,7 +259,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
 // CHECK: convert_dot_16_16_32_f8E4M3FNUZ_f8E4M3FNUZ_f32
     tt.func @convert_dot_16_16_32_f8E4M3FNUZ_f8E4M3FNUZ_f32(%a: tensor<16x32x!a_ty, #dot_operand_a>, %b: tensor<32x16x!b_ty, #dot_operand_b>) -> tensor<16x16x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<16x16x!c_ty, #blocked>
@@ -280,7 +280,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
 // CHECK: convert_dot_16_16_32_f8E4M3FNUZ_f8E5M2FNUZ_f32
     tt.func @convert_dot_16_16_32_f8E4M3FNUZ_f8E5M2FNUZ_f32(%a: tensor<16x32x!a_ty, #dot_operand_a>, %b: tensor<32x16x!b_ty, #dot_operand_b>) -> tensor<16x16x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<16x16x!c_ty, #blocked>
@@ -301,7 +301,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
 // CHECK: convert_dot_16_16_32_f8E5M2FNUZ_f8E4M3FNUZ_f32
     tt.func @convert_dot_16_16_32_f8E5M2FNUZ_f8E4M3FNUZ_f32(%a: tensor<16x32x!a_ty, #dot_operand_a>, %b: tensor<32x16x!b_ty, #dot_operand_b>) -> tensor<16x16x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<16x16x!c_ty, #blocked>
@@ -322,7 +322,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [16, 16], isTransposed = false}>
 // CHECK: convert_dot_16_16_32_f8E5M2FNUZ_f8E5M2FNUZ_f32
     tt.func @convert_dot_16_16_32_f8E5M2FNUZ_f8E5M2FNUZ_f32(%a: tensor<16x32x!a_ty, #dot_operand_a>, %b: tensor<32x16x!b_ty, #dot_operand_b>) -> tensor<16x16x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<16x16x!c_ty, #blocked>
@@ -343,7 +343,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
 // CHECK: convert_dot_4_4_64_f16_f16_f32
     tt.func @convert_dot_4_4_64_f16_f16_f32(%a: tensor<4x64x!a_ty, #dot_operand_a>, %b: tensor<64x4x!b_ty, #dot_operand_b>) -> tensor<4x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x4x!c_ty, #blocked>
@@ -364,7 +364,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
 // CHECK: convert_dot_4_4_64_bf16_bf16_f32
     tt.func @convert_dot_4_4_64_bf16_bf16_f32(%a: tensor<4x64x!a_ty, #dot_operand_a>, %b: tensor<64x4x!b_ty, #dot_operand_b>) -> tensor<4x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x4x!c_ty, #blocked>
@@ -385,7 +385,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
 // CHECK: convert_dot_4_4_64_f32_f32_f32
     tt.func @convert_dot_4_4_64_f32_f32_f32(%a: tensor<4x64x!a_ty, #dot_operand_a>, %b: tensor<64x4x!b_ty, #dot_operand_b>) -> tensor<4x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x4x!c_ty, #blocked>
@@ -406,7 +406,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 4], isTransposed = false}>
 // CHECK: convert_dot_4_4_64_i8_i8_i32
     tt.func @convert_dot_4_4_64_i8_i8_i32(%a: tensor<4x64x!a_ty, #dot_operand_a>, %b: tensor<64x4x!b_ty, #dot_operand_b>) -> tensor<4x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0> : tensor<4x4x!c_ty, #blocked>
@@ -503,7 +503,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
 // CHECK: convert_dot_64_4_4_f16_f16_f32
     tt.func @convert_dot_64_4_4_f16_f16_f32(%a: tensor<64x4x!a_ty, #dot_operand_a>, %b: tensor<4x4x!b_ty, #dot_operand_b>) -> tensor<64x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<64x4x!c_ty, #blocked>
@@ -524,7 +524,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
 // CHECK: convert_dot_64_4_4_bf16_bf16_f32
     tt.func @convert_dot_64_4_4_bf16_bf16_f32(%a: tensor<64x4x!a_ty, #dot_operand_a>, %b: tensor<4x4x!b_ty, #dot_operand_b>) -> tensor<64x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<64x4x!c_ty, #blocked>
@@ -545,7 +545,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
 // CHECK: convert_dot_64_4_4_f32_f32_f32
     tt.func @convert_dot_64_4_4_f32_f32_f32(%a: tensor<64x4x!a_ty, #dot_operand_a>, %b: tensor<4x4x!b_ty, #dot_operand_b>) -> tensor<64x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<64x4x!c_ty, #blocked>
@@ -566,7 +566,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [64, 4], isTransposed = false}>
 // CHECK: convert_dot_64_4_4_i8_i8_i32
     tt.func @convert_dot_64_4_4_i8_i8_i32(%a: tensor<64x4x!a_ty, #dot_operand_a>, %b: tensor<4x4x!b_ty, #dot_operand_b>) -> tensor<64x4x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0> : tensor<64x4x!c_ty, #blocked>
@@ -663,7 +663,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
 // CHECK: convert_dot_4_64_4_f16_f16_f32
     tt.func @convert_dot_4_64_4_f16_f16_f32(%a: tensor<4x4x!a_ty, #dot_operand_a>, %b: tensor<4x64x!b_ty, #dot_operand_b>) -> tensor<4x64x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x64x!c_ty, #blocked>
@@ -684,7 +684,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
 // CHECK: convert_dot_4_64_4_bf16_bf16_f32
     tt.func @convert_dot_4_64_4_bf16_bf16_f32(%a: tensor<4x4x!a_ty, #dot_operand_a>, %b: tensor<4x64x!b_ty, #dot_operand_b>) -> tensor<4x64x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x64x!c_ty, #blocked>
@@ -705,7 +705,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
 // CHECK: convert_dot_4_64_4_f32_f32_f32
     tt.func @convert_dot_4_64_4_f32_f32_f32(%a: tensor<4x4x!a_ty, #dot_operand_a>, %b: tensor<4x64x!b_ty, #dot_operand_b>) -> tensor<4x64x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0.000000e+00> : tensor<4x64x!c_ty, #blocked>
@@ -726,7 +726,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 :
 #dot_operand_a = #triton_gpu.dot_op<{opIdx=0, parent=#blocked}>
 #dot_operand_b = #triton_gpu.dot_op<{opIdx=1, parent=#blocked}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
-// CHECK: #mfma = #triton_gpu.mfma<{version = 3.0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
+// CHECK: #mfma = #triton_gpu.mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [1, 1], instrShape = [4, 64], isTransposed = false}>
 // CHECK: convert_dot_4_64_4_i8_i8_i32
     tt.func @convert_dot_4_64_4_i8_i8_i32(%a: tensor<4x4x!a_ty, #dot_operand_a>, %b: tensor<4x64x!b_ty, #dot_operand_b>) -> tensor<4x64x!c_ty, #blocked> {
         %cst_c = arith.constant dense<0> : tensor<4x64x!c_ty, #blocked>
